### ISSUE TITLE
Add support for screen layout in MS Power point

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#8025]
Dear all,

I would like to start a discussion on how we could make NVDA able to present a slide in power point as better as possible. It should give the user a better visualization of how elements are ordered on a slide.

My suggestion is to implement kind of screen layout similar to the sreen layout in browsers. Even if we cannot act with every element on a slide, an user would get a faster visualization of the message.

Example:
Yesterday <diagonal down arrow>
Today <right arrow> rainy
Tomorow <diagonal up arro>

By pressing down arrow, I can understand that all three day were rainy because of the arro symbols.
As of now, in power point I must navigate by tab and shift+tab and NVDA jumps between every shape on a slide. This makes it impossible to imagine what the purpose of the slide really is.

I must add that this should be an optional slide viewer since text placeholders are to comprehensive to be included in screen layout. But if we could navigate a slide by elements using arrow keys instead of tab and shift + tab, this would simplify the visualization also when complex text fields are included.

Comments are appreciated.

Thanks.

**Comment by leonardder** [comment]
Have you already tried to run a slide show (f5) and doesn't this solve your case? If not, it would be nice if you could elaborate on whether slide show view is not sufficient for you.

**Comment by Adriani90** [comment]
@Leonardder when pressing f5 NVDA concentrates on showing as much text as possible but you don't really get an information on where things are on the screen and which shape they have.

Let's take the example above: assume the days (yesterday, today and tomorrow) are rectangle shaped forms and "rainy" is a text field. When pressing f5, the arrows are not being reported and the shapes rectangle and text field neather.

I would expect following:
1. When pressing down arrow three times, NVDA would report
Rectangle yesterday <diagonal down arrow>
rectangle today <right arrow> text field rainy
rectangle tomorrow <diagonal up arrow>

Now when I come with the cursor on rectangle tomorrow and press enter, I can edit this shape. Pressing escape would finish the edit mode. When pressing something like strg+down arrow, the rectangle tomorrow would move downwards. Pressing right arrow would take me to the <diagonal up arrow> symbol and pressing context menu key would open the menu to copy that symbol and other actions. Pressing enter on the symbol <diagonal up arrow> would not do anything because <diagonal up arrow> cannot be edited. Pressing right arrow again would report "no element on this line anymore".

This approach would let users create good presentations and at the same time visualize the slide while editing.

In fact, NVDA should let me navigate the slide from left to right by elements and from up to down by lines, although lines would be quite abstract in this case because elements can have different size.

**Comment by Adriani90** [comment]
I have edited the arrow symbols.

**Comment by dkager** [comment]
So instead of supporting screen layout, you are saying that the shapes and possible decorations of objects on a slide should be announced better?

**Comment by Adriani90** [comment]
No, the shapes are correctly pronounced when editing a slide, my last reply to @Leonardder aimed to clarify that screen layout in presentation mode is not enough to visualize things on a slide. An user should be able to understand the order of elements on a slide while editing the slide. The presentation mode is not addressed by this issue at all.
